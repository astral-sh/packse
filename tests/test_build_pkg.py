import pytest

from .common import snapshot_command


def test_build_pkg_no_name(snapshot):
    assert snapshot_command(["build-pkg"]) == snapshot


def test_build_pkg_no_version(snapshot):
    assert snapshot_command(["build-pkg", "foo"]) == snapshot


def test_build_pkg_invalid_version(snapshot, tmpcwd):
    target = tmpcwd / "test.json"
    target.touch()
    assert snapshot_command(["build-pkg", "foo", "bar"]) == snapshot


@pytest.mark.usefixtures("tmpcwd")
def test_build_pkg(snapshot):
    assert (
        snapshot_command(
            ["build-pkg", "foo", "1.0.0"],
            snapshot_filesystem=True,
        )
        == snapshot
    )


@pytest.mark.usefixtures("tmpcwd")
def test_build_pkg_no_wheel(snapshot):
    assert (
        snapshot_command(
            ["build-pkg", "foo", "1.0.0", "--no-wheel"],
            snapshot_filesystem=True,
        )
        == snapshot
    )


@pytest.mark.usefixtures("tmpcwd")
def test_build_pkg_no_sdist(snapshot):
    assert (
        snapshot_command(
            ["build-pkg", "foo", "1.0.0", "--no-sdist"],
            snapshot_filesystem=True,
        )
        == snapshot
    )


@pytest.mark.usefixtures("tmpcwd")
def test_build_pkg_wheel_tags(snapshot):
    assert (
        snapshot_command(
            ["build-pkg", "foo", "1.0.0", "-t", "tag1", "--wheel-tag", "tag2"],
            snapshot_filesystem=True,
        )
        == snapshot
    )
