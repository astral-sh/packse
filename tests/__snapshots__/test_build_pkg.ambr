# serializer version: 1
# name: test_build_pkg
  dict({
    'exit_code': 0,
    'filesystem': dict({
      'build/foo/foo-1.0.0/.gitignore': '''
        __pycache__
        *.pyc
  
      ''',
      'build/foo/foo-1.0.0/pyproject.toml': '''
        [build-system]
        requires = ["hatchling"]
        build-backend = "hatchling.build"
        
        [tool.hatch.build.targets.wheel]
        packages = ["src/foo"]
        
        [tool.hatch.build.targets.sdist]
        only-include = ["src/foo"]
        
        [project]
        name = "foo"
        version = "1.0.0"
        dependencies = []
        requires-python = ">=3.8"
        description = ''''''
        
        [project.optional-dependencies]
  
      ''',
      'build/foo/foo-1.0.0/src/foo/__init__.py': '''
        __version__ = "1.0.0"
  
      ''',
      'dist/foo/foo-1.0.0-py3-none-any.whl': 'md5:3745cac160b20f05730d9f16cbafeaa7',
      'dist/foo/foo-1.0.0.tar.gz': 'md5:6837d63fd3c0d2d873cb1bf80b722540',
      'tree': '''
        test_build_pkg0
        ├── build
        │   └── foo
        │       └── foo-1.0.0
        │           ├── .gitignore
        │           ├── pyproject.toml
        │           └── src
        │               └── foo
        │                   └── __init__.py
        └── dist
            └── foo
                ├── foo-1.0.0-py3-none-any.whl
                └── foo-1.0.0.tar.gz
        
        7 directories, 5 files
  
      ''',
    }),
    'stderr': '''
      Generated project for 'foo-1.0.0'in [TIME]
      Built package 'foo-1.0.0' in [TIME]
      Linked distribution to dist/foo/foo-1.0.0-py3-none-any.whl
      Linked distribution to dist/foo/foo-1.0.0.tar.gz
      Done in [TIME]
  
    ''',
    'stdout': '',
  })
# ---
# name: test_build_pkg_invalid_version
  dict({
    'exit_code': 1,
    'stderr': '''
      Version 'bar' is not valid..
  
    ''',
    'stdout': '',
  })
# ---
# name: test_build_pkg_no_name
  dict({
    'exit_code': 2,
    'stderr': '''
      usage: packse build-pkg [-h] [-t WHEEL_TAG] [--no-wheel]
                              [--requires-python REQUIRES_PYTHON] [--no-sdist]
                              [--force] [-v] [-q]
                              name version
      packse build-pkg: error: the following arguments are required: name, version
  
    ''',
    'stdout': '',
  })
# ---
# name: test_build_pkg_no_sdist
  dict({
    'exit_code': 0,
    'filesystem': dict({
      'build/foo/foo-1.0.0/.gitignore': '''
        __pycache__
        *.pyc
  
      ''',
      'build/foo/foo-1.0.0/pyproject.toml': '''
        [build-system]
        requires = ["hatchling"]
        build-backend = "hatchling.build"
        
        [tool.hatch.build.targets.wheel]
        packages = ["src/foo"]
        
        [tool.hatch.build.targets.sdist]
        only-include = ["src/foo"]
        
        [project]
        name = "foo"
        version = "1.0.0"
        dependencies = []
        requires-python = ">=3.8"
        description = ''''''
        
        [project.optional-dependencies]
  
      ''',
      'build/foo/foo-1.0.0/src/foo/__init__.py': '''
        __version__ = "1.0.0"
  
      ''',
      'dist/foo/foo-1.0.0-py3-none-any.whl': 'md5:3745cac160b20f05730d9f16cbafeaa7',
      'tree': '''
        test_build_pkg_no_sdist0
        ├── build
        │   └── foo
        │       └── foo-1.0.0
        │           ├── .gitignore
        │           ├── pyproject.toml
        │           └── src
        │               └── foo
        │                   └── __init__.py
        └── dist
            └── foo
                └── foo-1.0.0-py3-none-any.whl
        
        7 directories, 4 files
  
      ''',
    }),
    'stderr': '''
      Generated project for 'foo-1.0.0'in [TIME]
      Built package 'foo-1.0.0' in [TIME]
      Linked distribution to dist/foo/foo-1.0.0-py3-none-any.whl
      Done in [TIME]
  
    ''',
    'stdout': '',
  })
# ---
# name: test_build_pkg_no_version
  dict({
    'exit_code': 2,
    'stderr': '''
      usage: packse build-pkg [-h] [-t WHEEL_TAG] [--no-wheel]
                              [--requires-python REQUIRES_PYTHON] [--no-sdist]
                              [--force] [-v] [-q]
                              name version
      packse build-pkg: error: the following arguments are required: version
  
    ''',
    'stdout': '',
  })
# ---
# name: test_build_pkg_no_wheel
  dict({
    'exit_code': 0,
    'filesystem': dict({
      'build/foo/foo-1.0.0/.gitignore': '''
        __pycache__
        *.pyc
  
      ''',
      'build/foo/foo-1.0.0/pyproject.toml': '''
        [build-system]
        requires = ["hatchling"]
        build-backend = "hatchling.build"
        
        [tool.hatch.build.targets.wheel]
        packages = ["src/foo"]
        
        [tool.hatch.build.targets.sdist]
        only-include = ["src/foo"]
        
        [project]
        name = "foo"
        version = "1.0.0"
        dependencies = []
        requires-python = ">=3.8"
        description = ''''''
        
        [project.optional-dependencies]
  
      ''',
      'build/foo/foo-1.0.0/src/foo/__init__.py': '''
        __version__ = "1.0.0"
  
      ''',
      'dist/foo/foo-1.0.0.tar.gz': 'md5:6837d63fd3c0d2d873cb1bf80b722540',
      'tree': '''
        test_build_pkg_no_wheel0
        ├── build
        │   └── foo
        │       └── foo-1.0.0
        │           ├── .gitignore
        │           ├── pyproject.toml
        │           └── src
        │               └── foo
        │                   └── __init__.py
        └── dist
            └── foo
                └── foo-1.0.0.tar.gz
        
        7 directories, 4 files
  
      ''',
    }),
    'stderr': '''
      Generated project for 'foo-1.0.0'in [TIME]
      Built package 'foo-1.0.0' in [TIME]
      Linked distribution to dist/foo/foo-1.0.0.tar.gz
      Done in [TIME]
  
    ''',
    'stdout': '',
  })
# ---
# name: test_build_pkg_wheel_tags
  dict({
    'exit_code': 0,
    'filesystem': dict({
      'build/foo/foo-1.0.0/.gitignore': '''
        __pycache__
        *.pyc
  
      ''',
      'build/foo/foo-1.0.0/pyproject.toml': '''
        [build-system]
        requires = ["hatchling"]
        build-backend = "hatchling.build"
        
        [tool.hatch.build.targets.wheel]
        packages = ["src/foo"]
        
        [tool.hatch.build.targets.sdist]
        only-include = ["src/foo"]
        
        [project]
        name = "foo"
        version = "1.0.0"
        dependencies = []
        requires-python = ">=3.8"
        description = ''''''
        
        [project.optional-dependencies]
  
      ''',
      'build/foo/foo-1.0.0/src/foo/__init__.py': '''
        __version__ = "1.0.0"
  
      ''',
      'dist/foo/foo-1.0.0-tag1.whl': 'md5:3745cac160b20f05730d9f16cbafeaa7',
      'dist/foo/foo-1.0.0-tag2.whl': 'md5:3745cac160b20f05730d9f16cbafeaa7',
      'dist/foo/foo-1.0.0.tar.gz': 'md5:6837d63fd3c0d2d873cb1bf80b722540',
      'tree': '''
        test_build_pkg_wheel_tags0
        ├── build
        │   └── foo
        │       └── foo-1.0.0
        │           ├── .gitignore
        │           ├── pyproject.toml
        │           └── src
        │               └── foo
        │                   └── __init__.py
        └── dist
            └── foo
                ├── foo-1.0.0-tag1.whl
                ├── foo-1.0.0-tag2.whl
                └── foo-1.0.0.tar.gz
        
        7 directories, 6 files
  
      ''',
    }),
    'stderr': '''
      Generated project for 'foo-1.0.0'in [TIME]
      Built package 'foo-1.0.0' in [TIME]
      Linked distribution to dist/foo/foo-1.0.0-tag1.whl
      Linked distribution to dist/foo/foo-1.0.0-tag2.whl
      Linked distribution to dist/foo/foo-1.0.0.tar.gz
      Done in [TIME]
  
    ''',
    'stdout': '',
  })
# ---
